onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -153,6 +154,21 @@ Status GraphPartitioner::Partition(Graph& graph, bool export_dll, FuncManager& f
     std::vector<Node*> nodes_need_compile;
     std::vector<std::unique_ptr<ComputeCapability>> capabilities =
         provider->GetCapability(graph_viewer, kernel_registry_mgr_.GetKernelRegistriesByProviderType(provider->Type()));
+
+    // For CUDA EP, exclude the subgraph that is prefered to be placed in CPU
+    // These are usually shape related computation subgraphs
+    // Following logic can be extended for other EPs
+    if (provider->Type() == kCudaExecutionProvider) {
```

Comment:
>provider->Type() == kCudaExecutionProvider [](start = 8, length = 42)

Would it better to add an interface to IExecutionProvider and let EP return its preferences of CPU execution?

Reply:
Closing this thread, as we decided to make this an Util function.